atomos.utils/atomos.utils.substrate.impl/src/main/java/org/apache/felix/atomos/utils/substrate/impl/NativeImageArgumentsImpl.java

```diff
@@ -121,7 +121,7 @@ public List<String> arguments()
             .map(Path::toAbsolutePath)//
             .sorted(NativeImageArgumentsImpl::byAbsolutePath)//
             .map(Path::toString)//
-            .collect(Collectors.joining(":"));
+            .collect(Collectors.joining(NativeImageCliUtil.isWindows() ? ";" : ":"));
```

Comment:
Just stumbled upon this and thought that `File.pathSeparator` should already have the distinction between windows and unix build in.

Reply:
It is not only the path separator that is different between windows and unix. It is also the separator in the classpath or modulepath that is a semicolon on windows but a colon on unix. That is a different place.

Reply:
Just to provide more details in case someone else stumbles across this in the future:

`java --help` on Windows returns

```
--class-path <class search path of directories and zip/jar files>
                  A ; separated list of directories, JAR archives,
                  and ZIP archives to search for class files.

--module-path <module path>...
                  A ; separated list of directories, each directory
                  is a directory of modules.
```

`java --help` on a Unix system returns:

```
--class-path <class search path of directories and zip/jar files>
                  A : separated list of directories, JAR archives,
                  and ZIP archives to search for class files.

--module-path <module path>...
                  A : separated list of directories, each directory
                  is a directory of modules.
```